marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/DhsDeployer.java

```diff
@@ -55,6 +57,26 @@ public void deployAsSecurityAdmin(HubConfigImpl hubConfig) {
         dhsDeployer.deploy(hubConfig.getAppConfig());
     }
 
+    /**
+     * Compares current ML version with the minimum required version and fails immediately
+     * if current ML version is lower than the minimum required version.
+     * @param hubConfig
+     */
+    public void checkCompatibility(HubConfig hubConfig) {
```

Comment:
I think this should be built into the deploy methods and not be a separate public method that clients have to call (which means you don't need to change the Gradle tasks). 

Let's give it a more descriptive name too - "check" methods are often vague about what they actually do or what the output will be. Our purpose here is to throw an exception if the ML version is invalid. So let's call the method that - "throwExceptionIsMarkLogicVersionIsInvalid". That leaves no doubt as to what the method will do.

Reply:
I did not add it to deploy methods because these methods are being used by certain tests that might not have the same version requirements. For instance, the version check condition does not account for the nightlies.